### PR TITLE
fix: charts not displaying when tree_type changed

### PR DESCRIPTION
**Pull request by hasnain2808**
Charts were not displaying on clicking the checkboxes in Sales and Purchase Analytics Reports
**Solved**
analytics report

### CODE REVIEW

File: erpnext/buying/report/purchase_analytics/purchase_analytics.js [hunk 1]
```diff
@@ -75,62 +75,66 @@ frappe.query_reports["Purchase Analytics"] = {
...
+					entry = {
+						name: row_name,
+						values: row_values,
+					};
 
 					let raw_data = frappe.query_report.chart.data;
 					let new_datasets = raw_data.datasets;
 
-					var found = false;
+					let found = false;
```
Comment (Anurag810) [inline erpnext/buying/report/purchase_analytics/purchase_analytics.js]: What is `found` please use a better variable name and also check other variable names.

File: erpnext/buying/report/purchase_analytics/purchase_analytics.js [hunk 2]
```diff
@@ -75,62 +75,66 @@ frappe.query_reports["Purchase Analytics"] = {
...
 
-					for(var i=0; i < new_datasets.length;i++){
-						if(new_datasets[i].name == row_name){
+					for (let i = 0; i < new_datasets.length; i++) {
```
Comment (Anurag810) [inline erpnext/buying/report/purchase_analytics/purchase_analytics.js]: use `for-each loop` for better readability.

**Review by Anurag810 — CHANGES_REQUESTED** [review]

**Review by Anurag810 — CHANGES_REQUESTED** [review]
Also, port to version-12

**Review by Anurag810 — CHANGES_REQUESTED** [review]
> Charts were not displaying on clicking the checkboxes in Sales and Purchase Analytics Reports
> **Solved**
>analytics report

Why It is showing two lines if only one row is checked. on switching tree type to **Item Group**

**Comment by hasnain2808** [comment]
> > Charts were not displaying on clicking the checkboxes in Sales and Purchase Analytics Reports
> > **Solved**
> >analytics report
> 
> Why It is showing two lines if only one row is checked. on switching tree type to **Item Group**

@Anurag810 
It is looking that way because the gif Is looping
Will upload the full video soon

**Comment by hasnain2808** [comment]
@Anurag810 
[Full Video](https://drive.google.com/file/d/1zKXlt0QJPGVxe--H3FuQqCnS3zlPOau7/view?usp=sharing)

**Review by Anurag810 — APPROVED** [review]

**Pull request merged**